jax/_src/lax/lax.py

```diff
@@ -3605,25 +3607,25 @@ def reducer_fn(op_val_index, acc_val_index):
 
 
 def _reduce_logical_shape_rule(operand, *, axes):
-  if operand.dtype != np.bool_:
-    msg = "logical reduction requires operand dtype bool, got {}."
+  if operand.dtype != np.bool_ and not np.issubdtype(operand.dtype, np.integer):
+    msg = "logical reduction requires operand dtype bool or int, got {}."
     raise TypeError(msg.format(operand.dtype))
```

Comment:
Since we're touching this code, we may as well update it to modern string formatting:
```python
  if operand.dtype != np.bool_ and not np.issubdtype(operand.dtype, np.integer):
    raise TypeError(f"logical reduction requires operand dtype bool or int, got {operand.dtype}.")
```